adapters/repos/db/lsmkv/compactor_set_skipkey_test.go

```diff
@@ -0,0 +1,128 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2025 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package lsmkv
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/require"
+	"github.com/weaviate/weaviate/entities/cyclemanager"
+)
+
+func TestCompactionSetSkipKeys(t *testing.T) {
+	maxSize := 5
+
+	keysToSkip := map[string]struct{}{}
+	ctx := t.Context()
+	opts := []BucketOption{
+		WithStrategy(StrategySetCollection),
+		WithForceCompaction(true),
+		WithShouldSkipKeyFunction(
+			func(key []byte, ctx context.Context) (bool, error) {
```

Comment:
could you define this function somewhere and then use it in the code AND the tests? Then we can be sure that we use the same function in both places